public/app/features/explore/state/query.ts

```diff
@@ -355,9 +340,34 @@ export const importQueries = (
  * @param modification Action object with a type, e.g., ADD_FILTER
  * @param modifier Function that executes the modification, typically `datasourceInstance.modifyQueries`.
  */
-export function modifyQueries(exploreId: ExploreId, modification: QueryFixAction, modifier: any): ThunkResult<void> {
-  return (dispatch) => {
-    dispatch(modifyQueriesAction({ exploreId, modification, modifier }));
+export function modifyQueries(
+  exploreId: ExploreId,
+  modification: QueryFixAction,
+  modifier: (query: DataQuery, modification: QueryFixAction) => Promise<DataQuery>
+): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    const state = getState().explore[exploreId];
+    if (state == null) {
```

Comment:
Why do we need to check if the state is `null`? Can it be `null` when this action is invoked?

Reply:
i don't know. the typescript-types allow for it, and i don't like adding `!` to workaround the type-safety 😿 . i added a comment there to explain it's to make typescript happy. what do you think?

Reply:
I agree `!` is not good if it's used as a workaround, but in the context of reducer actions it is fine because:
* all actions are dispatched only once the state is initialized (so at least `left` pane exists)
* all actions originate from the existing pane

We use the convention that the compiler cannot determine here. And I agree that maybe we could do a better job but we already use non-null assertion in various actions (just search for `getState().explore[exploreId]!`). For sake of consistency, I'd prefer to use it here as well. If we ever find a way to get rid of this crap it will be easier to find all use cases if they are consistent.

Reply:
i see. i adjusted the code 👍